factcast-store-validation/src/test/java/org/factcast/store/pgsql/internal/listen/PgListenerTest.java

```diff
@@ -0,0 +1,162 @@
+    PgConnectionSupplier ds;
+
+    @Mock
+    AsyncEventBus bus;
+
+    @Mock
+    PgConnection conn;
+
+    @Mock
+    PreparedStatement ps;
+
+    Predicate<Connection> tester = c -> true;
+
+    @Captor
+    ArgumentCaptor<PgListener> captor;
+
+    @SuppressWarnings("unchecked")
+    @Test
+    void testCheckFails() throws SQLException {
+
+        Mockito.when(ds.get()).thenReturn(conn);
+        Mockito.when(conn.prepareStatement(anyString())).thenReturn(ps);
+
+        tester = mock(Predicate.class);
+        Mockito.when(tester.test(any())).thenReturn(false, false, true, false);
+        PgListener l = new PgListener(ds, bus, tester);
+        l.afterPropertiesSet();
+        verify(bus, times(3)).post(any(FactInsertionEvent.class));
+        verifyNoMoreInteractions(bus);
+
+        l.destroy();
+    }
+
+    @Test
+    void testListen() throws Exception {
+        Mockito.when(ds.get()).thenReturn(conn);
+        Mockito.when(conn.prepareStatement(anyString())).thenReturn(ps);
+
+        PgListener l = new PgListener(ds, bus, tester);
+        l.afterPropertiesSet();
+        sleep(100);
+        verify(bus).post(any(FactInsertionEvent.class));
+        verifyNoMoreInteractions(bus);
+        verify(conn, atLeastOnce()).prepareStatement(PgConstants.LISTEN_SQL);
+        verify(ps, atLeastOnce()).execute();
+
+        l.destroy();
+    }
+
+    @Test
+    void testNotify() throws Exception {
+
+        Mockito.when(ds.get()).thenReturn(conn);
+        Mockito.when(conn.prepareStatement(anyString())).thenReturn(ps);
+
+        PgListener l = new PgListener(ds, bus, tester);
+        when(conn.getNotifications(anyInt())).thenReturn(new PGNotification[] { //
+                new Notification(PgConstants.CHANNEL_NAME, 1), //
+                new Notification(PgConstants.CHANNEL_NAME, 1), //
+                new Notification(PgConstants.CHANNEL_NAME, 1) }, new PGNotification[] {
+                        new Notification(PgConstants.CHANNEL_NAME, 2) }, //
+                new PGNotification[] { new Notification(PgConstants.CHANNEL_NAME, 3) }, null);
+        l.afterPropertiesSet();
+        sleep(400);
+        // 4 posts: one scheduled, 3 from notifications
+        verify(bus, times(4)).post(any(FactInsertionEvent.class));
+
+        l.destroy();
+    }
+
+    @Test
+    void testNotifyScheduled() throws Exception {
+
+        Mockito.when(ds.get()).thenReturn(conn);
+        Mockito.when(conn.prepareStatement(anyString())).thenReturn(ps);
+
+        PgListener l = new PgListener(ds, bus, tester);
+        when(conn.getNotifications(anyInt())).thenReturn(null);
+        l.afterPropertiesSet();
+        sleep(200);
+        // one scheduled
+        verify(bus, times(1)).post(any(FactInsertionEvent.class));
+
+        l.destroy();
+    }
+
+    @Test
+    void testStop() throws Exception {
+        Mockito.when(ds.get()).thenReturn(conn);
+        Mockito.when(conn.prepareStatement(anyString())).thenReturn(mock(PreparedStatement.class));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary use of fully qualified name 'Mockito.when' due to existing static import 'org.mockito.Mockito.when'](https://app.codacy.com/manual/uwe/factcast/pullRequest?prid=4877539)